adapters/access.go

```diff
@@ -197,11 +199,13 @@ func (a *AccessAdapter) GetTransactionResult(
 		return nil, convertError(err)
 	}
 
-	result.Events, err = ConvertCCFEventsToJsonEvents(result.Events)
-	if err != nil {
-		return nil, convertError(err)
+	// Convert CCF events to JSON events, else return CCF encoded version
+	if requiredEventEncodingVersion == entities.EventEncodingVersion_JSON_CDC_V0 {
+		result.Events, err = ConvertCCFEventsToJsonEvents(result.Events)
+		if err != nil {
+			return nil, convertError(err)
+		}
 	}
```

Comment:
The old and new code seem repeated in multiple places. Maybe make this a function to avoid the repeated code